.eslintrc.js

```diff
@@ -37,35 +37,6 @@ module.exports = {
 				selector: 'CallExpression[callee.name="deprecated"] Property[key.name="version"][value.value=/' + majorMinorRegExp + '/]',
 				message: 'Deprecated functions must be removed before releasing this version.',
 			},
-			{
```

Comment:
@aduth - this is the rule that causes ESLint to randomly throw errors when processing JSX code with the latest version of both Babel and ESLint. Any ideas about how to fix it?

Reply:
> @aduth - this is the rule that causes ESLint to randomly throw errors when processing JSX code with the latest version of both Babel and ESLint. Any ideas about how to fix it?

I think we avoided the error previously based on the fact that filtering to `Literal` would be applied before calling the `nth-child`, as it would occur when trying to access properties on specific types of nodes. I'm guessing something changed here with the order in how the filtering is applied.

I imagine if we were to move more of this implementation into JavaScript (i.e. not rely on this `esquery` AST selector syntax), it could work again. Not immediately sure how much effort might be involved, or if it's more worthwhile to consider moving it to its own proper custom rule in `@wordpress/eslint-plugin`.

Reply:
Tried a few ideas briefly, but no luck so far.

As far as I can tell, there's no way to include JavaScript in the rule to implement logic around considering the selected value in `no-restricted-syntax` (reasonable considering ESLint configurations are often JSON). The only options are `selector` and `message`.

I tried a variation which opted for separate rules to test ` > :not(ArrayExpression)` and ` > * + :not(ArrayExpression)` for first and second arguments respectively, but puzzlingly the same error occurs (despite not using `:nth-child`).

```js
{
	selector: (
		`CallExpression[callee.name=/^(${ [
			'property',
			'matchesProperty',
			'path',
		].join( '|' ) })$/] > :not(ArrayExpression)`
	),
	message: 'Always pass an array as the path argument',
},
{
	selector: (
		`CallExpression[callee.name=/^(${ [
			'invokeMap',
			'get',
			'has',
			'hasIn',
			'invoke',
			'result',
			'set',
			'setWith',
			'unset',
			'update',
			'updateWith',
		].join( '|' ) })$/] > * + :not(ArrayExpression)`
	),
	message: 'Always pass an array as the path argument',
},
```

Reply:
I'm fine to have this not considered a blocker for merge. The rule is good to have, but not critical, and can be reintroduced in the future.